### PR TITLE
add filter and sorting to network interfaces tab

### PR DESCRIPTION
**Pull request by avivtur**
## 📝 Description

> Add a brief description

## 🎥 Demo

network-interfaces

Signed-off-by: Aviv Turgeman <aturgema@redhat.com>

### CODE REVIEW

**Comment by avivtur** [comment]
@glekner @metalice @pcbailey @yaacov @hstastna @upalatucci please review

File: src/views/virtualmachines/details/components/network/utils/utils.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  const drive = Object.keys(interfacesTypes).find((ifaceType: string) => iface?.[ifaceType]);
+  return drive ?? NO_DATA_DASH;
+};
+
+export const getPrintableNetworkInterfaceType = (iface: V1Interface): string =>
+  interfacesTypes[getNetworkInterfaceType(iface)];
+
+export const filters: RowFilter[] = [
+  {
+    filterGroupName: 'Interface Type',
```
Comment (pcbailey) [inline src/views/virtualmachines/details/components/network/utils/utils.ts:42]: Shouldn't this be translated as well?

File: src/views/virtualmachines/details/components/network/NetworkInterfaceRow.tsx [hunk 2]
```diff
@@ -20,16 +21,16 @@ const NetworkInterfaceRow: React.FC<RowProps<NetworkInterfaceRowProps>> = ({
         {network.name}
       </TableData>
       <TableData id="model" activeColumnIDs={activeColumnIDs}>
-        {iface.model || '-'}
+        {iface.model || NO_DATA_DASH}
       </TableData>
       <TableData id="network" activeColumnIDs={activeColumnIDs}>
-        {network.pod ? 'Pod networking' : network.multus?.networkName || '-'}
+        {network.pod ? 'Pod networking' : network.multus?.networkName || NO_DATA_DASH}
```
[the remark this answers is not in the document]
Comment (avivtur) [reply src/views/virtualmachines/details/components/network/NetworkInterfaceRow.tsx:27]: you're right, nice catch :)

File: src/views/virtualmachines/details/components/network/utils/utils.ts (hunk 1 above)
Comment (avivtur) [reply src/views/virtualmachines/details/components/network/utils/utils.ts:42]: I guess you're right, but I wasn't really sure about this, @glekner @yaacov do you know by any chance?

Comment (avivtur) [reply src/views/virtualmachines/details/components/network/utils/utils.ts]: changed as suggested to have translation, if not needed will be reverted in follow-up PR, @pcbailey @glekner @yaacov

**Pull request merged**